### PR TITLE
Vir 31 update position

### PR DESCRIPTION
**Pull request by joonasyr**
- Added DTO model which containts fields that can be updated for the Position model
- Created UpdatePosition function which handles HTTP PUT requests to update an existing position with id
- Implemented the UpdatePosition method within the PositionRepository which uses Entity Framework no mark entry as modified and save it to database

UpdatePosition can be expanded in the future to support fields in addition to PaattymisPvm and LopetusPaatosNumero

https://tuni-team-hvtovuq0.atlassian.net/jira/software/projects/VIR/boards/1/backlog?selectedIssue=VIR-31

### CODE REVIEW

**Comment by joonasyr** [comment]
Tested with the following curl format:

$uri = "http://localhost:7071/api/positions/ID"  # Replace with your ID
$body = '{"PaattymisPvm": "2024-11-10T00:00:00Z", "LopetusPaatosNumero": "9876543210}' # Replace values with your own
$headers = @{ "Content-Type" = "application/json" }
Invoke-RestMethod -Uri $uri -Method Put -Body $body -Headers $headers

File: backend/src/Functions/Positions/UpdatePosition.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using static Virkarekisteri.Utils.DeserializeHelper;
+
+namespace Virkarekisteri.Functions.Positions;
+
+public class UpdatePosition
+{
+    private readonly ILogger<UpdatePosition> _logger;
+    private readonly PositionRepository _positionRepository;
+
+    public UpdatePosition(ILogger<UpdatePosition> logger, PositionRepository positionRepository)
```
Comment (0x464e) [inline backend/src/Functions/Positions/UpdatePosition.cs]: We can use primary constructor here for cleaner/less boilerplate. For an example see e.g. https://github.com/virkarekisteri/virkarekisteri/blob/fcc57a7ae3eace9669fc1b4c3d818c5b349aeba2/backend/src/Functions/Positions/CreatePosition.cs#L11

File: backend/src/Repositories/PositionRepository.cs [hunk 2]
```diff
@@ -29,4 +29,16 @@ public async Task<Position> CreatePosition(Position position)
         await db.SaveChangesAsync();
         return position;
     }
+
+    /// <summary>
+    /// Updates an existing Position in the database with the specified changes.
+    /// This method marks the Position as modified and saves the changes to the database.
+    /// </summary>
+    /// <param name="position">The Position object containing updated values to save.</param>
```
Comment (0x464e) [inline backend/src/Repositories/PositionRepository.cs]: Out of date param name in XML comment

File: backend/src/Models/UpdatePositionDto.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+namespace Virkarekisteri.Models
```
Comment (0x464e) [inline backend/src/Models/UpdatePositionDto.cs:1]: Should we separate DTOs to a different folder from the EF models?

File: backend/src/Models/UpdatePositionDto.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+{
+    public class UpdatePositionDto
+    {
+        public DateTime? PaattymisPvm { get; set; }
```
Comment (0x464e) [inline backend/src/Models/UpdatePositionDto.cs]: Lets use English. Use the same names as in the Position model

**Review by 0x464e — CHANGES_REQUESTED** [review]
Looks good apart from the minor things I commented

File: backend/src/Models/UpdatePositionDto.cs (hunk 3 above)
Comment (joonasyr) [reply backend/src/Models/UpdatePositionDto.cs:1]: Can be restructured later on if seen feasible.

**Comment by joonasyr** [comment]
After switching to english, the body of the request has to be in English as well. Here my curl for testing:

$uri = "http://localhost:7071/api/positions/id"  # Replace with your ID
$body = '{"EndedAt": "2020-11-10T00:00:00Z", "EndingDecisionNumber": "11111111"}'
$headers = @{ "Content-Type" = "application/json" }

Invoke-RestMethod -Uri $uri -Method Put -Body $body -Headers $headers

**Review by 0x464e — APPROVED** [review]

**Comment by 0x464e** [comment]
> After switching to english, the body of the request has to be in English as well. Here my curl for testing:
> 
> $uri = "http://localhost:7071/api/positions/id" # Replace with your ID $body = '{"EndedAt": "2020-11-10T00:00:00Z", "EndingDecisionNumber": "11111111"}' $headers = @{ "Content-Type" = "application/json" }
> 
> Invoke-RestMethod -Uri $uri -Method Put -Body $body -Headers $headers

As a tip, you can also conveniently use your IDEs HTTP client to send requests more conveniently.  

E.g. in Rider you can also quickly access it with:  
image

**Pull request merged**